job-application-tracker-api/src/applications/applications.service.ts

```diff
@@ -60,4 +66,20 @@ export class ApplicationsService {
         if (!deletedApplication) throw new NotFoundException('no such application');
         return ApplicationMapper.toApplicationDto(deletedApplication);
     }
+
+    async createState(appId:string,newState:ApplicationState, userId: string){
+        const application: Application = await this.applicationRepository.findById(appId);
+        if(!application || application.userId!==userId){
+            throw new NotFoundException('no such application');
+        }
+        application.states.push(newState);
+        const updatedApplication={
+            companyName: application.companyName,
+            position: application.position,
+            states:application.states,
+            userId:userId
+        };
+        const updateApplication = await this.applicationRepository.update(appId, updatedApplication);
```

Comment:
No need to create `updatedApplication`, just send `application` to `applicationRepository.update`